go/libraries/doltcore/table/editor/pk_table_editor.go

```diff
@@ -283,7 +285,7 @@ func (te *pkTableEditor) InsertRow(ctx context.Context, dRow row.Row) error {
 		if err != nil {
 			return err
 		}
-		return fmt.Errorf(ErrDuplicatePrimaryKeyFmt, keyStr)
+		return table.NewBadRow(dRow, fmt.Sprintf(ErrDuplicatePrimaryKeyFmt, keyStr))
```

Comment:
This doesn't work with Daylon's outstanding change to handle duplicate primary keys here. That change requires this method to return an error type defined in the go-mysql-server package, which means you can't wrap it like this.

Change the logic that examines this error, rather than changing the error type.